### PR TITLE
use external files in data explorer

### PR DESCRIPTION
**Pull request by colombod**
Can load the js from a uri instead of the local  endpoint

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
+
+            formatted.Should().Match("*getExtensionRequire('nteract','1.0.0')(['nteract/index'], (nteract) => {*");
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs:49]: `getExtensionRequire` really needs a better name.

Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs:49]: `Should().Contain()` is probably better here.

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
+
+            formatted.Should().Match("*getJsLoader*");
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs:93]: Let's rename `getJsLoader`.

Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs:93]: `Should().Contain()` is preferable. The output is more informative I think.

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+            var data = new[]
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs:179]: Why not `data.ToDisplayString(TabularDataFormatter.MimeType)`?

File: samples/ExtensionLab/ExtensionLab-nteract.ipynb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "#r \"nuget:Microsoft.DotNet.Interactive.ExtensionLab,*-*\""
+   ],
+   "outputs": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "source": [
+    "#!connect mssql --create-dbcontext True --kernel-name adventureworks \"Persist Security Info=False; Integrated Security=true; Initial Catalog=AdventureWorks2019; Server=localhost\""
```
[the remark this answers is not in the document]
Comment (colombod) [reply samples/ExtensionLab/ExtensionLab-nteract.ipynb]: done

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs (hunk 1 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs]: done

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs (hunk 2 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs]: this is changed to accommodate synapse

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs (hunk 3 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs:179]: I am testing the tabluarJsonString formatter, the open producing the full html code

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs (hunk 2 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs]: done

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**